src/core/namespace.c

```diff
@@ -1073,6 +1074,27 @@ static int mount_sysfs(const MountEntry *m) {
         return 1;
 }
 
+static bool mount_option_supported(const char *fstype, const char *key, const char *value) {
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        /* This function assumes support by default. Only if the fsconfig() call fails with -EINVAL/-EOPNOTSUPP
+         * will it report that the option/value is not supported. */
+
+        fd = fsopen(fstype, FSOPEN_CLOEXEC);
+        if (fd < 0) {
+                if (errno != ENOSYS)
+                        log_debug_errno(errno, "Failed to open superblock context for '%s': %m", fstype);
+                return true; /* If fsopen() fails for whatever reason, assume the value is supported. */
+        }
+
+        r = fsconfig(fd, FSCONFIG_SET_STRING, key, value, 0);
+        if (r < 0 && !IN_SET(errno, EINVAL, EOPNOTSUPP, ENOSYS))
+                log_debug_errno(r, "Failed to set '%s=%s' on '%s' superblock context: %m", key, value, fstype);
+
+        return r >= 0 || !IN_SET(errno, EINVAL, EOPNOTSUPP);
```

Comment:
i am a bit confused about the error numbers. down here you check for EINVAL/EOPNOTSUPP, after the fsconfig() you also allow EOPNOTSUPP, and up top for fsconfig() just ENOSYS?

shouldn't this just use ERRNO_NOT_SUPPORTED for the fsconfig() check, and then for the other two do `ERRNO_IS_NOTSUPPORTED(errno) || errno == EINVAL`?

Reply:
Using `ERRNO_IS_NOT_SUPPORTED()` means that if `fsconfig()` or `fsopen()` are not available, we will assume `hidepid=` is not supported. I think we want the opposite behavior, this functions should assume support if 'fsconfig()` or `fsopen()` are not available, since we can't actually do the check because the syscalls aren't there.

We don't log on `ENOSYS` because it's expected these syscalls won't be available everywhere so it seems like it'd be noisy if we logged on `ENOSYS` as well.